torch/_inductor/select_algorithm.py

```diff
@@ -2387,6 +2389,35 @@ def do_autotuning(choices, precompile_fn, hint_override: Optional[int] = None):
                 log.debug("Prescreening elapsed time: %.02fs", prescreening_elapse)
 
             autotune_start_ts = time.time()
+
+            if best_config_future is not None:
+                best_config = await_sync(best_config_future)
+
+                important_keys = [
+                    "ACC_TYPE",
+                    "ALLOW_TF32",
+                    "BLOCK_K",
+                    "BLOCK_M",
+                    "BLOCK_N",
+                    "EVEN_K",
+                    "GROUP_M",
+                    "USE_FAST_ACCUM",
+                    "num_stages",
+                    "num_warps",
+                    "num_consumer_groups",
+                    "num_buffers_warp_spec",
+                ]
```

Comment:
This is very similar to the cache workup that @laithsakka did. We had a sev internally because we weren't caching on the right keys. WIth @laithsakka's work we have a test where if any part of the template changes, you will get an error if the corresponding keys are not updated: 

see: https://github.com/pytorch/pytorch/pull/151773/files#diff-ee670915dba4dc797dbf884564b9170b16fbc81ba52153bf32e18d5675721eabR1096-R1111

Would it be possible to reuse that? And also - go through a common path as above ?

cc @jansel for thoughts

Reply:
I agree we should reuse that